File: J_se_les12/HW/Core/Presenter/PresenterBase.java
package Java.J_se_les12.HW.Core.Presenter;

import Java.J_se_les12.HW.Core.Logic.Interfaces.ILFactory;

public abstract class PresenterBase {
    ILFactory factory;

    protected PresenterBase(ILFactory factory) {
        this.factory = factory;

    }

    public abstract void doThis();

}

File: J_se_les13/HW/Core/Presenter/PresenterBase.cs
using HW.Core.Logic.Interfaces;
namespace HW.Core.Presenter
{
    public abstract class PresenterBase
    {
        public ILFactory Factory { get; set; }
        protected PresenterBase(ILFactory factory)
        {
            Factory = factory;
        }
        public abstract void DoThis();
    }
}